src/main/kotlin/com/devfloor/untitled/common/exception/NotFoundException.kt

```diff
@@ -0,0 +1,6 @@
+package com.devfloor.untitled.common.exception
+
+data class NotFoundException(
+    override val message: String?
+) : RuntimeException() {
+}
```

Comment:
이거 그냥 메시지를 파라미터로 받는 생성자 말고 기본 생성자도 추가해주면 좋을듯? NotFoundException(), NotFoundException(message: String?) 이렇게 두 개!

Reply:
이거 `EntityNotFoundException`으로 이름 하는건 어때? 별론가?? ㅋㅋㅋ

Reply:
> `EntityNotFoundException`

흐음 나쁘지 않은거같긴한데, 뭔가 우리가 DB에서 Entity를 찾는다고 말하기보다는 DB에서 `해당 아티클` 이나 `해당 유저` 혹은 `데이터`를 찾는다고 하잖아?? 그런 의미에서 Entity -> Instance나 Object, Data같은 다른것들은 어떤거같아??

Reply:
> 이거 그냥 메시지를 파라미터로 받는 생성자 말고 기본 생성자도 추가해주면 좋을듯? NotFoundException(), NotFoundException(message: String?) 이렇게 두 개!

이건 혹시 어떤 이유에서야??? 방금 생각해봤는데 
나는 오히려 Message를 nullable하지 않게 만들어서 어떠한 경우에도 메시지를 쓰도록 바꾸는게 더 좋을거같다고 생각했어 !
그리고 나중에 혹시나 메시지가 전혀 필요 없는 상황이 발생하면, 그 때 기본 생성자를 추가하는 것도 좋지 않나??!

```suggestion
    override val message: String
```

Reply:
1. Object나 Data의 경우 클래스 종류로 존재하니까 조금 애매할 거 같고 Instance도 괜찮은거 같기도 하고? 둘 중 하나 맘에 드는걸로 ㄲ
2. 아 이건 뭐 형 말도 맞는거 같고 뭘 고르든 상관없는 것 같아! 그냥 익셉션의 경우 한 번 만들면 거의 안 건드릴거 같아서 만드는 김에 만들어두면 좋을거 같아서? 
기본 생성자를 두면 나중에 람다식 사용할 때 편하다보니 쓰일지는 모르겠지만 주로 기본생성자랑 메세지를 받는 형식의 생성자 2개씩은 오버로딩 해두더라고. 그냥 이건 뭐 제안 정도라 안해도 상관없어!! ㅋㅋ
아 그런데 저거 nullable할 수 밖에 없을거 같은게 오버로딩이라 매개변수 타입 맞춰야 하지 않나? RuntimeException의 생성자 파라미터가 Nullable이라 자동으로 Nullable로 생성된 거 같은데?? 저거 고쳐져??

Reply:
1. 준형님 의견도 들어보면 좋을 듯 합니다 !
2. 일단 그렇다면 nullable하지 않게 고치고 나중에 필요에따라 추가하는거로 할까!? 
> nullable할 수 밖에 없을거 같은게 오버로딩이라 매개변수 타입 맞춰야 하지 않나? 

아 이거는 사실 내가 명시한거야, 어떻게 전달할지 명확히 몰랐어서 nullable해놨는데 오버라이딩이라 Nullable하지 않게 바꿀 수 있는듯 !

Reply:
근데 이 exception 이미 존재하는거였네..? 어카지? 상관없나 ?,,,

![image](https://user-images.githubusercontent.com/49271821/115367553-461e7400-a201-11eb-8859-12fc98787cb1.png)

Reply:
> 조회할때 공용으로 쓸지 전부 나눠서 쓸지? 나눠서 정의해서 사용한다면 인스턴스든 데이터든 구분만 되면 될거같아요!

1. 별 다른 차이점이 없을 것 같긴해서 같이 써도 될 것 같아요 ! 만약 비즈니스적으로 차이점을 두어야 한다면 나중에 나누는 것도 고려하면 좋을거같아요 ㅎㅎ

> 존재하는 exception 처리가 되는지 테스트를 해봐야 알거같아요!

2. 이 부분은 이미 존재하는 `InstanceNotFoundException` 이나 `EntityNotFoundException` 을 사용해도 되는지 테스트를 해봐야 할 것 같다 라는 의미로 이해해도 될까요?
이거에 대한 제 생각은, JPA에서 사용하기 위해 만든거같이 이미 존재하는 것을 사용해도 되는가?? 라는 의문이 있어서요 ! 만약 문제가 없다면 괜찮지만, 일단 저는 준형님이 올려주신 링크(비지니스Exception 상속받아서 구체화하는 것) 처럼 해도 좋을거같긴해요